src/app/components/core/developer/additional-services/additional-service-modal/additional-service-modal.component.ts

```diff
@@ -70,18 +75,31 @@ export class AdditionalServiceModalComponent {
     private dialogService: DialogService,
     private fileUploaderService: FileUploaderService,
     private configurationControlService: ConfigurationControlService,
-    public rolesService: RolesService) { }
+    private additionalServicesUtils: AdditionalServicesUtils,
+    public rolesService: RolesService) {
+      this.activatedRoute.queryParams.subscribe(params => {
+        if (params?.name) {
+            this.fromNavbar = true;
+            this.getServiceInfo(params, params?.pollingScheduleID);
+            // Wait to get html page loaded
+            setTimeout(() => {
+              this.toggleModal(true);
+            }, 500);
+        }
+      });
+    }
 
   ngOnInit() { }
 
-  getServiceInfo(serviceInfo, availableServicePkgs, pollingScheduleID) {
+  getServiceInfo(serviceInfo, pollingScheduleID, from = null) {
+    this.navigateFromParent = from;
     this.serviceName = serviceInfo.name ? serviceInfo.name : '';
-    this.isServiceEnabled = ["shutdown", "disabled", "installed", ""].includes(serviceInfo.state) ? false : true;
-    this.isServiceAvailable = serviceInfo.added;
+    this.isServiceEnabled = serviceInfo.isEnabled ? (serviceInfo.isEnabled === 'true' || serviceInfo.isEnabled === true) : false;
+    this.isServiceAvailable = serviceInfo.added ? (serviceInfo.added === 'true' || serviceInfo.added === true) : false;
```

Comment:
`(serviceInfo.added === 'true' || serviceInfo.added === true) `
It means serviceInfo.added can hold both type of values string & boolean? This is not correct it should either be boolean or string and should be handled while creating the serviceInfo object. 
same is true for 
`serviceInfo.isEnabled === 'true' || serviceInfo.isEnabled === true`

Reply:
This component is being called via route (not via child component) in the navbar. There is a reason behind it (I'll let you know) and from navbar this `serviceInfo` is being passed via `queryParams`. As you know, `queryParams` automatically converted into string. So I've to do it like above to get the values of variables.